pkg/kubelet/eviction/helpers_test.go

```diff
@@ -102,12 +104,23 @@ func TestGetReclaimableThreshold(t *testing.T) {
 				},
 			},
 		},
+		"no thresholds": {thresholds: []evictionapi.Threshold{}, shouldNotHaveFound: true},
+		"threshold was crossed but reclaim not implemented (invalid signal)": {
+			thresholds: []evictionapi.Threshold{
+				{
+					Signal: "mem.available",
+				},
+			},
+			shouldNotHaveFound: true,
+		},
 	}
 	for testName, testCase := range testCases {
 		sort.Sort(byEvictionPriority(testCase.thresholds))
 		_, _, ok := getReclaimableThreshold(testCase.thresholds)
-		if !ok {
+		if !ok && !testCase.shouldNotHaveFound {
 			t.Errorf("Didn't find reclaimable threshold, test: %v", testName)
+		} else if ok && testCase.shouldNotHaveFound {
+			t.Errorf("Shouldn't have found reclaimable threshold, test: %v", testName)
 		}
```

Comment:
It's not very clear what will happen in other cases, e.g. when ok is True, but shouldNotHavefound is false. Can you cleanup the code a little bit?

Reply:
resolved in 47a74aebe8b